### PR TITLE
Add thunderhub app

### PR DESCRIPTION
**Pull request by apotdevin**
# App Submission

### App name
ThunderHub

### Version
v0.12.1

### One line description of the app
A simple and intuitive UI to take full control of your Lightning Node

### Summary of the app
ThunderHub allows you to take full control of your Lightning Node with a slick and awesome UI. Explore all the options that ThunderHub has to offer, from sending and receiving Lightning payments, checking your nodes health statistics and even more advanced options like channel rebalancing and multi-path lightning payments. Managing and monitoring your node has never been easier with transaction reports, graphs and a huge assortment of different features all bundled inside of this great tool.

### Developed by
@apotdevin

### Developer website
https://apotdevin.com/

### Source code repository
https://github.com/apotdevin/thunderhub

### Support link
https://t.me/thunderhub

### Uses
- [ ] Bitcoin Core
- [ ] Electrum server
- [X] LND

### 256x256 SVG icon
ThunderHub

### App screenshots
NightLight
Forwards
Channels

### I have tested my app on:
- [x] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [ ] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

File: apps/thunderhub/docker-compose.yml
Comment (AaronDewes) [inline apps/thunderhub/docker-compose.yml:27]: You should add compatibility with the current Umbrel URL.
```suggestion
      LOGOUT_URL: ${DEVICE_HOSTS:-"http://umbrel.local"}
```

File: apps/thunderhub/docker-compose.yml
Comment (AaronDewes) [inline apps/thunderhub/docker-compose.yml:25]: It accesses the mount point inside the container, so you can hardcode these paths.
```suggestion
      SSO_MACAROON_PATH: "/lnd/data/chain/bitcoin/${BITCOIN_NETWORK}"
      SSO_CERT_PATH: "/lnd/tls.cert"
```

File: apps/thunderhub/docker-compose.yml
Comment (AaronDewes) [inline apps/thunderhub/docker-compose.yml:10]: This should always be called `web` (was pointed out in the developer chat)
```suggestion
  web:
```

File: apps/thunderhub/docker-compose.yml
Comment (AaronDewes) [inline apps/thunderhub/docker-compose.yml:11]: This should be pinned to a specific sha-256

**Review by AaronDewes — CHANGES_REQUESTED** [review]

**Review by AaronDewes — DISMISSED** [review]

File: apps/thunderhub/docker-compose.yml
Comment (mayankchhabra) [reply apps/thunderhub/docker-compose.yml]: `$DEVICE_HOSTS` is an array of all the possible URLs and IP addresses that the user could be accessing their Umbrel at. It wouldn't exactly tell you which URL the user is currently at (umbrel.local, Tor hidden service, or IP address) plus we don't expose this variable to apps (as it can leak the dashboard's hidden service address).

It would perhaps be better to drop the redirect logic entirely, 'cause even though you can get the IP address/hostname via `window.hostname` in the browser on local network to correctly figure out the right redirect URL, you won't be able to figure out dashboard's onion address if the user is accessing via Tor.

File: apps/thunderhub/docker-compose.yml
Comment (mayankchhabra) [reply apps/thunderhub/docker-compose.yml]: While the image should be pinned to the SHA-256 digest of it's multi-arch manifest list, we don't expect the app developers to do that. We'll pin it ourselves as a part of the final changes before merge:

> After you've submitted your app, we'll review your pull request, create the required Tor hidden services for it, make some adjustments in the docker-compose.yml file, such as removing any port conflicts with other apps, pinning Docker images to their sha256 digests, assigning unique IP addresses to the containers, etc before merging. 
>
> https://github.com/getumbrel/umbrel/blob/master/apps/README.md

File: apps/thunderhub/data/thubConfig.yaml
Comment (louneskmt) [inline apps/thunderhub/data/thubConfig.yaml:4]: Hey! I think you should use the env variable LND_IP et LND_PORT instead of hard coding the host:port. You can for example set a container env variable for the server url:
```
environment:
    - SERVER_URL: $LND_IP:$LND_PORT
```
```suggestion
    serverUrl: '${SERVER_URL}'
```
Or just set the same env var and use
```suggestion
    serverUrl: '${LND_IP}:${LND_PORT}'
```

Comment (apotdevin) [reply apps/thunderhub/data/thubConfig.yaml:4]: Hey! Yeah plan on adding the functionality to read env vars inside the config file in a next release but it's currently not possible so had to hard code them

**Review by lukechilds — APPROVED** [review]
Awesome work @apotdevin!

Tested and working on both umbrel-dev and my mainnet Umbrel.

I've wired up the Tor hidden service and pinned the Docker Hub image.

I also changed the container to run as UID 1000 since it was running as root.

Those hardcoded connection details really do need to be set dynamically at some point too. It'll break if we ever change the container IPs which we likely will. The plan is that they will be assigned dynamically soon. But no need to let that block merging this.

Great stuff!

**Comment by apotdevin** [comment]
Fixed the hardcoded value in the yaml config file 🥳

**Review by lukechilds — APPROVED** [review]
Re-tested the latest changes and pinned the Docker image.

Thanks for resolving that so quickly!

**Review by mayankchhabra — APPROVED** [review]
This is just incredible work, @apotdevin! Tried it out on my testnet Umbrel and it works flawlessly on both local network and Tor. 

image

**Comment by lukechilds** [comment]
Merged!

Congrats on being the first third-party app integrated into Umbrel @apotdevin!

**Pull request merged**
